LeaderboardBackend/Models/Entities/BaseEntity.cs

```diff
@@ -0,0 +1,38 @@
+using System.ComponentModel.DataAnnotations;
+using NodaTime;
+
+/// <summary>
+///     Base class that provides common fields to all Entities.
+/// </summary>
+public class BaseEntity
+{
+	/// <summary>
+	///     The time this entity was created.
+	/// </summary>
+	[Required]
+	public Instant CreatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was updated.
+	/// </summary>
+	[Required]
+	public Instant UpdatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was deleted.
+	/// </summary>
+	public Instant? DeletedAt { get; set; }
+
+	public override bool Equals(object? obj)
```

Comment:
This is wrong isn't it? Shouldn't equality be ID-based?

Reply:
Since there isn't an `Id` field here, I naturally couldn't match by ID. But as I've mentioned in Discord, I'm thinking about adding the `Id` field in here too, since all entities will have an ID column. And by having equality be ID-based, do you mean that it should only match by ID, or match by ID + other fields?

That being said; as you may be able to tell, each entity's `Equals` overrides match a varying number of fields. Not sure on how I should implement those either. It'd be great if you could take a look at those and advise too.

Reply:
Also ye I'll implement this change in #128

Reply:
For Runs we have a Guid and for everthing else we have an Id as an unique identifier. So either we change Run to use a Id or we leave it as it is now.

Reply:
I'm gonna change Runs to have longs as IDs